pylint/extensions/docparams.py

```diff
@@ -115,6 +115,9 @@ def check_functiondef_params(self, node, node_doc):
             node_doc, node.args, node, node_allow_no_param)
 
     def check_functiondef_returns(self, node, node_doc):
+        if utils.inside_yields(node):
```

Comment:
For Google and Numpy docstrings it's valid to have a yields section. Therefore this method should still raise a `redundant-return-doc` warning if the docstring type supports a "yield" section and the docstring defines a "return" section for a generator.
Does this make sense? I'm not sure if I've explained that well.